packages/editor/src/store/private-selectors.js

```diff
@@ -22,28 +27,41 @@ const EMPTY_INSERTION_POINT = {
  *
  * @return {Object} The root client ID, index to insert at and starting filter value.
  */
-export const getInsertionPoint = createRegistrySelector(
-	( select ) => ( state ) => {
-		if ( typeof state.blockInserterPanel === 'object' ) {
-			return state.blockInserterPanel;
-		}
+export const getInsertionPoint = createRegistrySelector( ( select ) =>
+	createSelector(
+		( state ) => {
+			if ( typeof state.blockInserterPanel === 'object' ) {
+				return state.blockInserterPanel;
+			}
 
-		if ( getRenderingMode( state ) === 'template-locked' ) {
+			if ( getRenderingMode( state ) === 'template-locked' ) {
+				const [ postContentClientId ] =
+					select( blockEditorStore ).getBlocksByName(
+						'core/post-content'
+					);
+				if ( postContentClientId ) {
+					return {
+						rootClientId: postContentClientId,
+						insertionIndex: undefined,
+						filterValue: undefined,
+					};
+				}
+			}
+
+			return EMPTY_INSERTION_POINT;
+		},
+		( state ) => {
 			const [ postContentClientId ] =
 				select( blockEditorStore ).getBlocksByName(
 					'core/post-content'
```

Comment:
Worth noting that this can lead to deoptimization: as long as there is a valid `state.blockInserterPanel`, the original selector never selected from `blockEditorStore` and didn't create a subcription to it. But the memoized version does always does the selection when calculation dependants.

There's probably no way around this other than abandoning the `getInsertionPoint` selector and inlining the code into the only place where it's used.

Reply:
Thanks for the feedback, @jsnajdr! We can't remove `getInsertionPoint` because of BC. We can stop using it in the core, but that just avoids the issue.